tfx/utils/kube_utils.py

```diff
@@ -187,3 +190,71 @@ def get_current_kfp_pod(client: k8s_client.CoreV1Api) -> k8s_client.V1Pod:
     return client.read_namespaced_pod(name=pod_name, namespace=namespace)
   except KeyError:
     raise RuntimeError('Cannot determine KFP pod from the environment.')
+
+
+def get_pod(core_api: k8s_client.CoreV1Api, pod_name: Text,
+            namespace: Text) -> Optional[k8s_client.V1Pod]:
+  """Get a pod from Kubernetes metadata API.
+
+  Args:
+    core_api: Client of Core V1 API of Kubernetes API.
+    pod_name: The name of the POD.
+    namespace: The namespace of the POD.
+
+  Returns:
+    The found POD object. None if it's not found.
+
+  Raises:
+    RuntimeError: When it sees unexpected errors from Kubernetes API.
+  """
+  try:
+    return core_api.read_namespaced_pod(name=pod_name, namespace=namespace)
+  except k8s_client.rest.ApiException as e:
+    if e.status != 404:
+      raise RuntimeError('Unknown error! \nReason: %s\nBody: %s' %
+                         (e.reason, e.body))
+    return None
+
+
+def wait_pod(core_api: k8s_client.CoreV1Api,
```

Comment:
Could you please rebase this once you are done with the other PRs, since I think this is shared.

Reply:
Right, I think it would be best to rebase this once the Kubernetes Dag Runner is merged, since it contains a lot of changes in kube_utils.